protocol/app/prepare/other_msgs.go

```diff
@@ -46,6 +54,16 @@ func RemoveDisallowMsgs(ctx sdk.Context, decoder sdk.TxDecoder, txs [][]byte) []
 
 	var filteredTxs [][]byte
 	for i, txBytes := range txs {
+		// If tx is index 0 and VE enabled, this ts is a VE tx and should be decoded with the extCommitCodec.
+		if i == constants.OracleInfoIndex && ve.VoteExtensionsEnabled(ctx) {
```

Comment:
can there be a situation where the VE is enabled but the first tx is not oracle VE info? for example, what happens when there's no oracle update because validators didn't add VEs?

Reply:
The expected behavior of the dydx PrepareProposal handlers is that the first tx of a block (after vote-extensions are enabled) will always unmarshal to an `extendedCommitInfo`, if validators broadcast empty vote-extensions in the previous block (assume ves were enabled 2 blocks ago), the first tx will contain an extendedCommitInfo containing the empty vote-extensions.